File: frontend/src/api/production.ts
import client from './client';

export interface ProductionStage {
  id: number;
  production_order_id: number;
  stage_name: string;
  status: string;
  start_time?: string;
  end_time?: string;
  progress: number;
  remarks?: string;
}

export interface ProductionOrder {
  id: number;
  order_id: number;
  status: string;
  assigned_workshop?: string;
  planned_start?: string;
  planned_end?: string;
  remarks?: string;
  created_at: string;
}

export interface ProductionDashboard {
  total_orders: number;
  in_progress: number;
  completed: number;
  delayed: number;
}

export const productionApi = {
  dashboard: () =>
    client.get<ProductionDashboard>('/production/dashboard'),

  listOrders: (params?: { skip?: number; limit?: number; status?: string }) =>
    client.get<ProductionOrder[]>('/production/orders', { params }),

  getOrder: (id: number) =>
    client.get<ProductionOrder & { stages: ProductionStage[] }>(`/production/orders/${id}`),

  createOrder: (data: {
    order_id: number;
    assigned_workshop?: string;
    planned_start?: string;
    planned_end?: string;
    remarks?: string;
    stages: { stage_name: string }[];
  }) =>
    client.post<ProductionOrder>('/production/orders', data),

  updateStage: (orderId: number, stageId: number, status: string, progress?: number) =>
    client.put(`/production/orders/${orderId}/stages/${stageId}`, null, { params: { status, progress } }),
};

File: backend/app/schemas/production.py
from datetime import datetime
from pydantic import BaseModel


class ProductionStageBase(BaseModel):
    stage_name: str
    status: str = "pending"
    start_time: datetime | None = None
    end_time: datetime | None = None
    progress: float = 0
    remarks: str | None = None


class ProductionStageCreate(ProductionStageBase):
    pass


class ProductionStageResponse(ProductionStageBase):
    id: int
    production_order_id: int

    class Config:
        from_attributes = True


class ProductionOrderBase(BaseModel):
    order_id: int
    assigned_workshop: str | None = None
    planned_start: datetime | None = None
    planned_end: datetime | None = None
    remarks: str | None = None


class ProductionOrderCreate(ProductionOrderBase):
    stages: list[ProductionStageCreate]


class ProductionOrderUpdate(BaseModel):
    status: str | None = None
    assigned_workshop: str | None = None
    planned_start: datetime | None = None
    planned_end: datetime | None = None
    remarks: str | None = None


class ProductionOrderResponse(ProductionOrderBase):
    id: int
    status: str
    created_at: datetime

    class Config:
        from_attributes = True


class ProductionOrderDetailResponse(ProductionOrderResponse):
    stages: list[ProductionStageResponse] = []


class ProductionDashboard(BaseModel):
    total_orders: int
    in_progress: int
    completed: int
    delayed: int
